src/Controller/ExerciseController.php

```diff
@@ -47,21 +56,13 @@ public function new(Request $request): Response
         ]);
     }
 
-    /**
-     * @Route("/{id}", name="exercise_show", methods={"GET"})
-     */
-    public function show(Exercise $exercise): Response
-    {
-        return $this->render('exercise/show.html.twig', [
-            'exercise' => $exercise,
-        ]);
-    }
-
     /**
      * @Route("/{id}/edit", name="exercise_edit", methods={"GET","POST"})
      */
     public function edit(Request $request, Exercise $exercise): Response
     {
+        $this->restrictModifyAccess($exercise);
+
```

Comment:
@daria-sieroshtan 
Blameable again.
Plan B would be to add a controller annotation to check this.
Doing the check explicitly in the code is a bad idea because you need this check in many places